node/podcontroller.go

```diff
@@ -263,6 +262,21 @@ func (pc *PodController) Run(ctx context.Context, podSyncWorkers int) (retErr er
 	}
 	log.G(ctx).Info("Pod cache in-sync")
 
+	pods, err := pc.podsLister.List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("failed to list pod error: %v", err)
+	}
+	var podCount int32
+	started := false
```

Comment:
At a minimum this will trigger a race condition.

Reply:
Yes, but actually would not influent the  final results.

I  have  added another commit based on this one. But I still think that would make the `add func` heavier.

WDYT? @sargun @cpuguy83